frontend/cypress/integration/common/istio_config.ts

```diff
@@ -519,31 +519,56 @@ Then('the AuthorizationPolicy should have a {string}', function (healthStatus: s
   ).hasCssVar('color', `--pf-v5-global--${healthStatus}-color--100`);
 });
 
+function waitUntilConfigIsVisible(attempt: number, crdInstanceName: string, crdName: string, namespace: string): void {
+  if (attempt === 0) {
+    return;
+  }
+  cy.request('GET', `${Cypress.config('baseUrl')}/api/istio/config?refresh=0`);
+  cy.get('[data-test="refresh-button"]').click();
+  ensureKialiFinishedLoading();
+  let found = false;
+  cy.get('tr')
+    .each($row => {
+      console.log($row);
+      const dataTestAttr = $row[0].attributes.getNamedItem('data-test');
+      if (dataTestAttr !== null) {
+        if (dataTestAttr.value === `VirtualItem_Ns${namespace}_${crdName.toLowerCase()}_${crdInstanceName}`) {
+          found = true;
+        }
+      }
+    })
+    .then(() => {
+      if (!found) {
+        cy.wait(40000);
+        waitUntilConfigIsVisible(attempt - 1, crdInstanceName, crdName, namespace);
```

Comment:
@matejnesuta does the tests eventually stop looping if there is error?

Reply:
Hello. Yes, there is a condition at the beginning of the function.

Reply:
oh I see now, thanks!